### PR TITLE
Make testExpiredPriceSet more realistic

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Identified while looking into building priceSets multiple times per #19469. The current `testExpiredPriceSet()` is not really an accurate test because it's not properly building the contribution page. This improves the test so that it builds and runs the contributionPage in the same way it would actually be used.

Before
----------------------------------------
Setting parameters directly, not really using a contribution page to test.

After
----------------------------------------
Using contributionPage form methods to build/test.

Technical Details
----------------------------------------

Comments
----------------------------------------
@eileenmcnaughton Would you be able to look at this. I've used `preProcess()` and `buildQuickForm()` methods directly so this should be a much more accurate test now.

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@mattwire you can likely call $form->preProcess in the unit test to do this stuff - the forms seem to support us doing that in the test much better than they used to now if you use the getFormObject method to get the form

File: tests/phpunit/CRM/Contribute/Form/Contribution/MainTest.php [hunk 1]
```diff
@@ -80,40 +80,44 @@ public function testSetRecurFunctionNotAvailable() {
...
+    ]));
+
+    /** @var \CRM_Contribute_Form_Contribution_Main $form */
+    $form = $this->getFormObject('CRM_Contribute_Form_Contribution_Main');
+    $contributionPage = reset($this->contributionPageCreate($contributionPageParams)['values']);
+    $form->set('id', $contributionPage['id']);
+    // PriceSetId gets looked up by CRM_Price_BAO_PriceSet::getFor() in ContributionBase::preProcess so we need to
+    // set it here:
+    CRM_Core_DAO::executeQuery("INSERT INTO civicrm_price_set_entity (entity_table, entity_id, price_set_id)
+            VALUES ('civicrm_contribution_page', {$contributionPage['id']}, {$params['priceSetID']})");
```
Comment (mattwire) [inline tests/phpunit/CRM/Contribute/Form/Contribution/MainTest.php:111]: I'm not sure how/when this is set on a real contribution page. There is no API for price_set_entity but this is required for the contributionPage to load the priceSet.

Comment (seamuslee001) [reply tests/phpunit/CRM/Contribute/Form/Contribution/MainTest.php:111]: @mattwire maybe https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/ContributionPage/Amount.php#L568

Comment (mattwire) [reply tests/phpunit/CRM/Contribute/Form/Contribution/MainTest.php]: Thanks, that looks like the one!

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Pull request merged**